pkg/prometheus/promcfg_test.go

```diff
@@ -5510,6 +5521,263 @@ scrape_configs:
+  - source_labels:
+    - __meta_kubernetes_endpoint_address_target_kind
+    - __meta_kubernetes_endpoint_address_target_name
+    separator: ;
+    regex: Node;(.*)
+    replacement: ${1}
+    target_label: node
+  - source_labels:
+    - __meta_kubernetes_endpoint_address_target_kind
+    - __meta_kubernetes_endpoint_address_target_name
+    separator: ;
+    regex: Pod;(.*)
+    replacement: ${1}
+    target_label: pod
+  - source_labels:
+    - __meta_kubernetes_namespace
+    target_label: namespace
+  - source_labels:
+    - __meta_kubernetes_service_name
+    target_label: service
+  - source_labels:
+    - __meta_kubernetes_pod_name
+    target_label: pod
+  - source_labels:
+    - __meta_kubernetes_pod_container_name
+    target_label: container
+  - action: drop
+    source_labels:
+    - __meta_kubernetes_pod_phase
+    regex: (Failed|Succeeded)
+  - source_labels:
+    - __meta_kubernetes_service_name
+    target_label: job
+    replacement: ${1}
+  - target_label: endpoint
+    replacement: web
+  - source_labels:
+    - __address__
+    target_label: __tmp_hash
+    modulus: 1
+    action: hashmod
+  - source_labels:
+    - __tmp_hash
+    regex: $(SHARD)
+    action: keep
+  keep_dropped_targets: %d
+  metric_relabel_configs: []
+`
+
+	for _, tc := range []struct {
+		version                    string
+		enforcedKeepDroppedTargets int
+		keepDroppedTargets         int
+		expected                   string
+	}{
+		{
+			version:                    "v2.46.0",
+			enforcedKeepDroppedTargets: -1,
+			keepDroppedTargets:         -1,
+			expected:                   expectNoKeepDroppedTargets,
+		},
+		{
+			version:                    "v2.47.0",
+			enforcedKeepDroppedTargets: -1,
+			keepDroppedTargets:         -1,
+			expected:                   expectNoKeepDroppedTargets,
+		},
+		{
+			version:                    "v2.46.0",
+			enforcedKeepDroppedTargets: 1000,
+			keepDroppedTargets:         -1,
+			expected:                   expectNoKeepDroppedTargets,
+		},
+		{
+			version:                    "v2.47.0",
+			enforcedKeepDroppedTargets: 1000,
+			keepDroppedTargets:         -1,
+			expected:                   fmt.Sprintf(expectKeepDroppedTargets, 1000),
+		},
+		{
+			version:                    "v2.46.0",
+			enforcedKeepDroppedTargets: 1000,
+			keepDroppedTargets:         2000,
+			expected:                   expectNoKeepDroppedTargets,
+		},
+		{
+			version:                    "v2.47.0",
+			enforcedKeepDroppedTargets: 1000,
+			keepDroppedTargets:         2000,
+			expected:                   fmt.Sprintf(expectKeepDroppedTargets, 1000),
+		},
+		{
+			version:                    "v2.46.0",
+			enforcedKeepDroppedTargets: 1000,
+			keepDroppedTargets:         500,
+			expected:                   expectNoKeepDroppedTargets,
+		},
+		{
+			version:                    "v2.47.0",
+			enforcedKeepDroppedTargets: 1000,
+			keepDroppedTargets:         500,
+			expected:                   fmt.Sprintf(expectKeepDroppedTargets, 500),
```

Comment:
Ok, I can see that fmt.Sprintf makes it difficult to use golden files.

I'd be interested to know what @nicolastakashi thinks about this situation.

Reply:
I'd still use golden files even if it means having more files in the testdata/ directory.

Reply:
I will update that. Other limit tests were not using golden file I had used one of them to create this one.

Reply:
updated